lib/api/decorators/schema.rb

```diff
@@ -0,0 +1,128 @@
+                                                                    name: title,
+                                                                    required: required,
+                                                                    writable: writable)
+          schema.min_length = min_length if min_length
+          schema.max_length = max_length if max_length
+
+          property property,
+                   getter: -> (*) { schema },
+                   writeable: false
+        end
+
+        def schema_with_allowed_link(property,
+                                          type: make_type(property),
+                                          title: make_title(property),
+                                          href_callback:,
+                                          required: true,
+                                          writable: true)
+          raise ArgumentError if property.nil?
+
+          property property,
+                   exec_context: :decorator,
+                   getter: -> (*) {
+                     representer = ::API::Decorators::AllowedValuesByLinkRepresenter.new(
+                       type: type,
+                       name: title,
+                       required: required,
+                       writable: writable)
+
+                     if represented.defines_assignable_values?
+                       representer.allowed_values_href = instance_eval(&href_callback)
+                     end
+
+                     representer
+                   }
+        end
+
+        def schema_with_allowed_collection(property,
+                                                type: make_type(property),
+                                                title: make_title(property),
+                                                values_callback:,
+                                                value_representer:,
+                                                link_factory:,
+                                                required: true,
+                                                writable: true)
+          raise ArgumentError unless property
+
+          property property,
+                   exec_context: :decorator,
+                   getter: -> (*) {
+                     representer = ::API::Decorators::AllowedValuesByCollectionRepresenter.new(
+                       type: type,
+                       name: title,
+                       current_user: current_user,
+                       value_representer: value_representer,
+                       link_factory: -> (value) { instance_exec(value, &link_factory) },
+                       required: required,
+                       writable: writable)
+
+                     if represented.defines_assignable_values?
+                       representer.allowed_values = instance_exec(&values_callback)
+                     end
+
+                     representer
+                   }
+        end
```

Comment:
I don't quite see your plan.

The major part of both methods **is** actually the getter block, so what you are saying is basically "the methods are equal except most of their content".

I also don't see how I would do it, because both methods also differ in their signature.

So when extracting a private method I would basically end up with a the same methods as before, but I don't need to set `exec_context: :decorator`.